### PR TITLE
Implement support for workspace metadata

### LINKED ISSUE #24: Support workspaces without the root package
The workspaces without a root package should just read the packages from `workspace.metadata.patch` instead of `package.metadata.patch`.

Patches are workspace-wide, so this, I'd say, is a more sensible thing to do either way.

### PR DESCRIPTION
**Pull request by MOZGIII**
Closes #24.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -303,18 +301,26 @@ fn main() -> Result<()> {
     let (pkg_set, resolve) = resolve_ws(&workspace)?;
 
     let mut patched = false;
+
+    let mut handle_custom_metadata =
```
Comment (mettke) [inline src/main.rs:305]: If possible, I would prefer this to be in its own function

**Review by mettke — CHANGES_REQUESTED** [review]
Very nice - thanks. Looks very good in general. Would like to see one change though

File: src/main.rs (hunk 1 above)
Comment (MOZGIII) [reply src/main.rs:305]: Sure, we'll just have to pass the resolver as an argument it seems; the way it's done in here is quite a bit cleaner.
I think I have an idea on how to recompose it even better though... Shall I try?

Comment (mettke) [reply src/main.rs:305]: Please go ahead. Looking forward to merge.

**Review by mettke — APPROVED** [review]
Nice I like it

**Comment by MOZGIII** [comment]
I've had to do a couple of force-pushes to fix some minor issues; the code can be further transformed in this style to reduce the ephemeral `Vec`s usage - but I'm not sure it's this PR is a good place to do it.

Maybe we should change the code back (omit the last commit), merge this one as it was, and create a new PR with a refactor?

**Comment by mettke** [comment]
Sounds good. Let do it like you said

**Pull request merged**